LiskHQ/lisk-hub
# Implement send LSK with Hardware Wallet Trezor Model T - Closes #1845 #1849 #1847

### What issue have I solved?

-- #1845 
-- #1847 
-- #1849 

### How have I implemented/fixed it?
For this implementation the same code for send LSK, using HW Ledger Nano S worked with the only difference that for the user of Trezor Model T you need to use the Trezor device ID and for Ledger you need to get the device ID from the key.

In summary for make this to work I had to update the TrezorLogin component to properly update the account reducer to provide the device ID, and with this change now I can use the same functionality as before but using the Trezor ID.


### How has this been tested?

1. Run the app with Electron
2. Open the developers mode and then add to the local storage one more item with key = trezor and value = true, after this refresh the app.
2. Login using the Trezor model T
3. Select the Trezor account
4. Go to Wallet -> Send LSK
5. Type any LSK address/account and amount
6. Go to the confirm screen after press the confirm button
7. Then here you should check the Trezor device for accept the transaction or cancel it
8. Then the final screen will show the result of the transaction

### Review checklist
- The PR follows our [Test guide](/LiskHQ/lisk-hub/blob/development/docs/TEST_GUIDE.md)
- The PR follows our [CSS guide](/LiskHQ/lisk-hub/blob/development/docs/CSS_GUIDE.md)
